engine/include/Assets.h

```diff
@@ -114,6 +115,42 @@ namespace nc
         std::string leftPath;
     };
 
+    enum class DescriptorType : uint8_t
```

Comment:
Looking for advice on where to put these. They can't easily go into ShaderAssetManager.h because it would cause circular dependencies, but they shouldn't bloat this guy like they are.

Reply:
I'd leave them here for now. We're going to end up with an include/assets/ directory, at which point we can start breaking this stuff into separate files where necessary.